### ISSUE TITLE
if remote file is missing, download function will create the local file and keep it open after the 550 error is thrown

### ISSUE THREAD

**Issue by SharpBCD** [patrickjuchli/basic-ftp#96]
I'm downloading csv files with names that include date. If the remote file does not exist I catch the 550 error and move on. However, the client.download function will create the local file with 0 bytes and the file handler remains open as it is never closed due to the error.

**Comment by patrickjuchli** [comment]
There are a few separate points here.

First, the download function doesn't create a file, you are. Probably with `fs.createWriteStream`? The download function of this library only knows about a writable stream, it doesn't know anything about files. That implies that it can't remove a file, you have to do that.

But you're right that the library should close the writable stream if an error occurred. In your case that would at least close the file handler as well.

Should this stream be closed with an error? This library always reports an error by raising an exception. If I close streams with errors you'd also have to attach error handlers to those. You'd then have to handle errors in two places: For every stream and for exceptions raised. I don't think that is a good idea and we should stick to a single way how errors are reported.

Final thought: Maybe the upload/download API should support file paths directly. The library would then be in charge of files created and could clean them up if necessary.

**Comment by SharpBCD** [comment]
Hi Patrick.
First of all thanks for writing this library. I first use the 'ftp' library which is not updated in 5 years, has no async/await, just event emitters and it failed when I made something a bit mode complex than 1 single file download. Your library saved me.

You are right, the function I provide creates the file. However, whenever I tried to do anything else than the bellow standard:
`await ftpClient.download(fs.createWriteStream(localFileFullName), remotePath)`
I got an error with `... .once in not a function` so this was the only implementation that worked. Maybe I did not dig into it enough but I made it work so it's fine.

I totally agree that two places to handle errors is absurd. 
If I can make a suggestions: how about calling the callback function after you got the first bytes from the remote file? So whatever errors that may occur before being able to write something will be addressed by the err handler with no other complication?

Your final thoughts are the best idea however. This library is remarkably simple to use so why not make it even simpler? Download (and upload) can be straight forward: sourcePath, destinationPath. Now it's like that anyway, I just added a function around destinationPath.

P.S. I implemented a workaround by listing the files and checking if the one I'm seeking exist before calling download. It works but it's not elegant, of course.
